OpenRA.Mods.Common/Traits/World/EditorActionManager.cs

```diff
@@ -114,6 +114,11 @@ public void Forward(int id)
 			while (undoStack.Peek().Id != id)
 				Redo();
 		}
+
+		public bool HasUnsavedItems()
+		{
+			return !(undoStack.Peek().Action is OpenMapAction || undoStack.Peek().Action is SaveMapAction);
```

Comment:
should also check that the `redoStack` is empty

Reply:
What scenario would that account for?

Reply:
When undoing some changes back to the Opened or Saved action, At least in Photoshop you will get a prompt warning about unsaved changes.

Reply:
Oh right I see. I'll do that now.